src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2408,19 +2388,18 @@ ProcessHavingClauseForWorkerQuery(Node *originalHavingQual,
 		 * If the GROUP BY or PARTITION BY is not on the distribution column
 		 * then we need to combine the aggregates in the HAVING across shards.
 		 */
-		WorkerAggregateWalkerContext *workerAggContext = palloc0(
-			sizeof(WorkerAggregateWalkerContext));
+		WorkerAggregateWalkerContext workerAggContext = { 0 };
```

Comment:
Maybe do this instead:
```c
WorkerAggregateWalkerContext workerAggContext = {
    .extendedOpNodeProperties = extendedOpNodeProperties
};

Reply:
I'm confused now. I'd done that previously, but it's been replaced with `{ 0 }` & I was seeing you were putting memset calls to zero structs beforehand

Reply:
Those were some special cases, that were flagged by static analysis. It's not necessary everywhere. The reason that they were flagged was that the the struct would later be returned by the function. This has the possible security issue that uninitiliazed stack memory leaves the current function, possibly ending up in some output. 

Here that the struct is not returned, so it's not necessary and we can use nice initializer syntax instead.